src/client.js

```diff
@@ -39,18 +39,28 @@ import './styles/main.pcss'
     }))
     // Read data from the API
     emitter.on('load', url => {
-      if (window.location.pathname + window.location.search !== url) {
+      if (window.location.pathname + window.location.search + window.location.hash !== url) {
         window.history.pushState(null, null, url)
+        window.dispatchEvent(new window.HashChangeEvent("hashchange"))        
         window.dispatchEvent(new window.PopStateEvent('popstate'))
       }
     })
+
     // Find data from the API
     emitter.on('getOptions', ({term, types, callback}) => api.find(term, types, callback))
     // Log in to the API
     emitter.on('login', () => api.login())
     // Log out of the API
     emitter.on('logout', () => api.logout())
 
+    window.addEventListener("hashchange", () => {
+      const hash = window.location.hash.substr(1)
```

Comment:
We should probably check if hash is not empty to ensure we are not using an empty query selector.